### PR TITLE
fix issue #220, using treemap

### PR DESCRIPTION
**Pull request by jeromevdl**
*Issue #, if available:* #220

*Description of changes:* Using TreeMap (and CASE_INSENSITIVE_ORDER initialisation) within a custom Header object.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by msailes** [comment]
Has this been tested on the AWS platform?

**Comment by jeromevdl** [comment]
> Has this been tested on the AWS platform?

Yes, LB and API GW were tested, I removed websockets

**Comment by msailes** [comment]
Thanks! @jeromevdl

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/APIGatewayProxyRequestEvent.java [hunk 1]
```diff
@@ -951,7 +953,15 @@ public APIGatewayProxyRequestEvent withHttpMethod(String httpMethod) {
...
-        this.headers = headers;
+        if (headers == null || headers.isEmpty()) {
+            this.headers = null;
+            return;
+        }
+
+        if (this.headers == null) {
+            this.headers = new HttpHeaders<>();
+        }
+        this.headers.putAll(headers);
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/APIGatewayProxyRequestEvent.java:964]: This is a breaking change for consumers using `setHeaders` and expecting it to override all previous values

**Comment by msailes** [comment]
Closing, replaced with https://github.com/aws/aws-lambda-java-libs/pull/298

**Pull request closed without merging**